### PR TITLE
clickable note card

### PR DESCRIPTION
**Pull request by foodstr**

### CODE REVIEW

**Review by BlowaterNostr — COMMENTED** [review]
This is working but will be refactored in about a month because of the design choices you made here.

File: UI/message-panel.tsx [hunk 1]
```diff
@@ -562,7 +570,31 @@ function NoteCard(
...
+                <div
+                    class={tw`px-4 my-1 py-2 border-2 border-[${PrimaryTextColor}4D] rounded-lg py-1 hover:bg-[${HoverButtonBackgroudColor}] cursor-pointer flex`}
+                    onClick={() => {
+                        let root: NostrEvent;
+                        root = event;
+                        if (event.parsedTags.root) {
+                            const res = eventSyncer.syncEvent(NoteID.FromHex(event.parsedTags.root[0]));
+                            if (res instanceof Promise) {
+                                return;
+                            }
```
Comment (BlowaterNostr) [inline UI/message-panel.tsx:582]: business logic like this should be handled in `update`

File: UI/message-thread-panel.tsx [hunk 2]
```diff
@@ -106,14 +109,36 @@ function MessageThreadBoxGroup(props: {
...
+    const shouldScroll = localStorage.getItem("ShouldScroll");
     const vnode = (
-        <ul class={tw`py-2`}>
+        <ul
+            class={tw`py-2`}
+            ref={async (e) => {
+                if (e) {
+                    if (focusedID && shouldScroll && shouldScroll == "true") {
+                        localStorage.removeItem("ShouldScroll");
+                        await sleep(200); // wait it finished loading
```
Comment (BlowaterNostr) [inline UI/message-thread-panel.tsx:124]: why? Is this logic relevant at all?

**Review by BlowaterNostr — CHANGES_REQUESTED** [review]

File: UI/message-panel.tsx (hunk 1 above)
Comment (foodstr) [reply UI/message-panel.tsx:582]: ok

File: UI/message-thread-panel.tsx (hunk 2 above)
Comment (foodstr) [reply UI/message-thread-panel.tsx:124]: those logic is for scrolling into view

Comment (BlowaterNostr) [reply UI/message-thread-panel.tsx:124]: Can it be separated to another PR?

**Comment by BlowaterNostr** [comment]
@foodstr please address the conflicts

File: UI/message-thread-panel.tsx (hunk 2 above)
Comment (foodstr) [reply UI/message-thread-panel.tsx:124]: ok

File: UI/app_update.ts [hunk 3]
```diff
@@ -364,6 +366,40 @@ export async function* UI_Interaction_Update(
...
             }
             model.rightPanelModel.show = true;
+        } else if (event.type == "ViewNoteDetail") {
+            let root: NostrEvent;
+            root = event.event;
+            if (event.event.parsedTags.root) {
+                const res = model.app.eventSyncer.syncEvent(NoteID.FromHex(event.event.parsedTags.root[0]));
+                if (res instanceof Promise) {
+                    continue;
+                }
```
Comment (BlowaterNostr) [inline UI/app_update.ts:376]: why `continue`?

File: UI/app_update.ts [hunk 4]
```diff
@@ -364,6 +366,40 @@ export async function* UI_Interaction_Update(
...
+                root = res;
+            }
+
+            if (root.kind == NostrKind.TEXT_NOTE) {
+                model.navigationModel.activeNav = "Social";
+                model.social.focusedContent = root;
+            } else if (model.navigationModel.activeNav == "DM") {
```
Comment (BlowaterNostr) [inline UI/app_update.ts:389]: why not `root.kind == NostrKind.Directed_Message`?

File: UI/message-panel.tsx [hunk 5]
```diff
@@ -68,13 +68,19 @@ export type DirectMessagePanelUpdate =
         show: boolean;
     }
     | ViewThread
-    | ViewUserDetail;
+    | ViewUserDetail
+    | ViewNoteDetail;
```
Comment (BlowaterNostr) [inline UI/message-panel.tsx:72]: Rename to `ViewNoteThread`

File: UI/message-thread-panel.tsx [hunk 6]
```diff
@@ -16,6 +17,8 @@ import { Editor, EditorEvent, EditorModel } from "./editor.tsx";
 import { Database_Contextual_View } from "../database.ts";
 import { ProfilesSyncer, UserInfo } from "./contact-list.ts";
 import { EventSyncer } from "./event_syncer.ts";
+import { DividerBackgroundColor, PrimaryBackgroundColor } from "./style/colors.ts";
+import { sleep } from "https://raw.githubusercontent.com/BlowaterNostr/csp/master/csp.ts";
```
Comment (BlowaterNostr) [inline UI/message-thread-panel.tsx:21]: Please always remember to remove unused imports

**Review by BlowaterNostr — CHANGES_REQUESTED** [review]

File: UI/app_update.ts [hunk 7]
```diff
@@ -364,6 +366,40 @@ export async function* UI_Interaction_Update(
                 }
             }
             model.rightPanelModel.show = true;
+        } else if (event.type == "ViewNoteDetail") {
+            let root: NostrEvent;
+            root = event.event;
```
[the remark this answers is not in the document]
Comment (foodstr) [reply UI/app_update.ts:371]: 图片

[the remark this answers is not in the document]
Comment (foodstr) [reply UI/app_update.ts:371]: because the eventsyncer can sync more then PlainText_Nostr_Event

File: UI/app_update.ts (hunk 3 above)
Comment (foodstr) [reply UI/app_update.ts:376]: Because if the no root event we can sync, I want to stop the ui render at this time

File: UI/app_update.ts (hunk 4 above)
Comment (foodstr) [reply UI/app_update.ts:389]: right

File: UI/message-panel.tsx (hunk 5 above)
Comment (foodstr) [reply UI/message-panel.tsx:72]: right

File: UI/app_update.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (BlowaterNostr) [reply UI/app_update.ts:371]: You can use type annotation in 1 line as well

**Comment by BlowaterNostr** [comment]
- favor #176

**Pull request closed without merging**